src/dft/descriptor.cxx

```diff
@@ -26,51 +27,126 @@ namespace detail {
         case config_param::FORWARD_SCALE: values_.fwd_scale = va_arg(vl, double); break;
         case config_param::BACKWARD_SCALE: values_.bwd_scale = va_arg(vl, double); break;
         case config_param::NUMBER_OF_TRANSFORMS:
             values_.number_of_transforms = va_arg(vl, int64_t);
             break;
-        case config_param::FWD_DISTANCE: values_.fwd_dist = va_arg(vl, int64_t); break;
-        case config_param::BWD_DISTANCE: values_.bwd_dist = va_arg(vl, int64_t); break;
+        case config_param::FWD_DISTANCE: values_.fwd_dist = va_arg(vl, std::int64_t); break;
+        case config_param::BWD_DISTANCE: values_.bwd_dist = va_arg(vl, std::int64_t); break;
         case config_param::PLACEMENT: values_.placement = va_arg(vl, config_value); break;
         case config_param::COMPLEX_STORAGE:
             values_.complex_storage = va_arg(vl, config_value);
             break;
+        case config_param::REAL_STORAGE:
+            throw mkl::unimplemented("DFT", "set_value", "Real storage not implemented.");
+            break;
         case config_param::CONJUGATE_EVEN_STORAGE:
             values_.conj_even_storage = va_arg(vl, config_value);
             break;
-        default: err = 1;
+        case config_param::ORDERING:
+            throw mkl::unimplemented("DFT", "set_value", "Ordering not implemented.");
+            break;
+        case config_param::TRANSPOSE:
+            throw mkl::unimplemented("DFT", "set_value", "Transpose not implemented.");
+            break;
+        case config_param::PACKED_FORMAT:
+            throw mkl::unimplemented("DFT", "set_value", "Packed format not implemented.");
+            break;
+        case config_param::COMMIT_STATUS:
+            throw mkl::invalid_argument("DFT", "set_value", "Read-only parameter.");
+            break;
+        default: throw mkl::invalid_argument("DFT", "set_value", "Invalid config_param argument.");
     }
     va_end(vl);
 }
+
 template <precision prec, domain dom>
 descriptor<prec, dom>::descriptor(std::vector<std::int64_t> dimensions)
         : dimensions_(std::move(dimensions)),
           rank_(dimensions.size()) {
-    // Compute default strides.
-    std::vector<std::int64_t> defaultStrides(rank_, 1);
-    for (int i = rank_ - 1; i < 0; --i) {
-        defaultStrides[i] = defaultStrides[i - 1] * dimensions_[i];
+    if (dimensions_.size() == 0) {
+        throw mkl::invalid_argument("DFT", "descriptor", "Cannot have 0 dimensional DFT.");
+    }
+    for (const auto& dim : dimensions) {
+        if (dim <= 0) {
+            throw mkl::invalid_argument("DFT", "descriptor",
+                                        "Invalid dimension value (negative or 0).");
+        }
+    }
+    // Compute default strides - see MKL C interface developer reference for CCE format.
+    std::vector<std::int64_t> strides(rank_ + 1, 1);
+    // The first variable stide value is different.
+    if (rank_ > 1) {
+        strides[rank_ - 1] = dimensions_[rank_] / 2 + 1;
```

Comment:
It is different because of the CCE packed data format - this is not the default of the C MKL library, but is the only packed data format for the OneAPI MKL.

I've found and corrected the error (I think) in [a152c39](https://github.com/hjabird/oneMKL/pull/3/commits/a152c39b277981ec04c0d2d8b584178a6c8930b9). The `dimensions_[rank_]` was indexing outside the dimensions array.

Reply:
I've corrected this further in [7d8a07b](https://github.com/hjabird/oneMKL/pull/3/commits/7d8a07b84e9470aa7c812e0c95278f5992130145) - since (as you said) it depend on the domain. I'm still not sure this is correct due - this aspect of the DFT implementation is confusing.

Reply:
Revised to match the Intel MKL in [f4ec8e5](https://github.com/hjabird/oneMKL/pull/3/commits/f4ec8e57ad5129244e2f6f84dcc19e9fde8230ad).